src/track.cpp

```diff
@@ -77,25 +77,39 @@ void Track::open(shared_ptr<DtlsSrtpTransport> transport) {
 #endif
 
 bool Track::outgoing(message_ptr message) {
-	auto direction = mMediaDescription.direction();
-	if (direction == Description::Direction::RecvOnly ||
-	    direction == Description::Direction::Inactive)
-		throw std::runtime_error("Track media direction does not allow sending");
 
-	if (mIsClosed)
-		throw std::runtime_error("Track is closed");
+    if (mRtcpHandler) {
+        message = mRtcpHandler->outgoing(message);
+        if (!message)
+            return false;
+    }
+
+    auto direction = mMediaDescription.direction();
+    if ((direction == Description::Direction::SendOnly ||
+         direction == Description::Direction::Inactive) &&
+        message->type != Message::Control) {
+        PLOG_WARNING << "Track media direction does not allow reception, dropping";
+    }
+
+	return sendControl(message);
+}
 
-	if (message->size() > maxMessageSize())
-		throw std::runtime_error("Message size exceeds limit");
+bool Track::sendControl(message_ptr message) {
```

Comment:
When I wrote this, it definitely felt like a hack. I need RtcpSession to be able to send data without calling itself again in the process.

Reply:
I think so, actually I have added the outgoing callback precisely for that reason, see below.